packages/app/src/components/UnauthorizedAccessModal.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { Button, Modal, Stack, Table, Text } from '@mantine/core';
+
+import { useHealthCheck } from '@/hooks/useMetadata';
+
+// Modal which will run health checks and display a popup if the user
+// has incorrect access permissions to the database.
+export const UnauthorizedAccessModal = () => {
+  const { results, refetch, isLoading } = useHealthCheck();
+
+  const unauthorizedResults = results.filter(r => r?.isAuthError);
+  const hasUnauthorized = unauthorizedResults.length > 0;
+
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <Modal
+      opened={hasUnauthorized ?? false}
+      centered
+      withCloseButton={false}
+      // An uncloseable modal
```

Comment:
Does the Team change control still work? I could see being part of multiple teams but one of them no longer has access to the db.  We don't necessarily want to prevent switching.

Reply:
Nice catch, that was broken. I have fixed this by adding a 'force close' feature. Switching teams reloads the page so if the other team has issues the modal will re-appear